tdrs-backend/tdpservice/parsers/validators.py

```diff
@@ -23,276 +24,361 @@ def validator(value):
 
 def or_validators(*args, **kwargs):
     """Return a validator that is true only if one of the validators is true."""
-    return lambda value: (True, None) if any([validator(value)[0] for validator in args])\
-        else (False, ' or '.join([validator(value)[1] for validator in args]))
+    return (
+        lambda value: (True, None)
+        if any([validator(value)[0] for validator in args])
+        else (False, " or ".join([validator(value)[1] for validator in args]))
+    )
 
 
 def and_validators(validator1, validator2):
     """Return a validator that is true only if both validators are true."""
-    return lambda value: (True, None) if (validator1(value)[0] and validator2(value)[0])\
-        else (False, (validator1(value)[1]) if validator1(value)[1] is not None else '' +
-              ' and ' + validator2(value)[1] if validator2(value)[1] is not None else '')
+    return (
+        lambda value: (True, None)
+        if (validator1(value)[0] and validator2(value)[0])
+        else (
+            False,
+            (validator1(value)[1])
+            if validator1(value)[1] is not None
+            else "" + " and " + validator2(value)[1]
+            if validator2(value)[1] is not None
+            else "",
+        )
+    )
+
 
 def extended_and_validators(*args, **kwargs):
     """Return a validator that is true only if all validators are true."""
-    return lambda value: (True, None) if all([validator(value)[0] for validator in args])\
-        else (False, ''.join([' and ' + validator(value)[1] if validator(value)[0] else '' for validator in args]))
+    return (
+        lambda value: (True, None)
+        if all([validator(value)[0] for validator in args])
+        else (
+            False,
+            "".join(
+                [
+                    " and " + validator(value)[1] if validator(value)[0] else ""
+                    for validator in args
+                ]
+            ),
+        )
+    )
```

Comment:
Can we please improve the readability and style here? The use of lambdas, nested if-elses, AND list comprehensions is non-pythonic and I imagine would be difficult to debug. It worked as a lambda with minimal logic that fit concisely into one line but we've boiled over on what fits well in that format I feel.

Something akin to this:
```
def extended_and_validators():
  return_value = False
  if (....):
     ....
  else:
      ....
  
  return return_value
 
```